targets/unity-v2/source/Authentication/PlayFabAuthService.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+using PlayFab.Authentication.Strategies;
+using PlayFab.ClientModels;
+using UnityEngine;
+using EntityKey = PlayFab.AuthenticationModels.EntityKey;
+
+namespace PlayFab
+{
+    /// <summary>
+    /// Supported Authentication types
+    /// Note: Add types to there to support more AuthTypes
+    /// See - https://api.playfab.com/documentation/client#Authentication
+    /// </summary>
+    public enum AuthTypes
+    {
+        None,
+        Silent,
+        EmailAndPassword,
+        UsernameAndPassword,
+        GooglePlayGames,
+        Facebook,
+        FacebookInstantGames,
+        GameCenter,
+        Kongregate,
+        NintendoSwitch,
+        XBoxLive,
+        PlayStation,
+        OpenId,
+        WindowsHello,
+        Twitch,
+        Steam
+    }
+
+    public class PlayFabAuthService
+    {
+        // Events to subscribe to for this service
+        public delegate void DisplayAuthenticationEvent();
+        public event DisplayAuthenticationEvent OnDisplayAuthentication;
+
+        public delegate void LoginSuccessEvent(LoginResult success);
+        public event LoginSuccessEvent OnLoginSuccess;
+
+        public delegate void PlayFabErrorEvent(PlayFabError error);
+        public event PlayFabErrorEvent OnPlayFabError;
+
+        public delegate void PlayFabLink(AuthTypes authType, PlayFabError error = null);
+        public event PlayFabLink OnPlayFabLink;
+        public event PlayFabLink OnPlayFabUnlink;
+
+        public string Email;
+        public string Username;
+        public string Password;
+        public GetPlayerCombinedInfoRequestParams InfoRequestParams;
+        public bool ForceLink;
+        
+        public string AuthTicket;
+        public string OpenIdConnectionId;
+        public string WindowsHelloChallengeSignature;
+        public string WindowsHelloPublicKeyHint;
+        
+        public string PlayFabId { get; private set; }
+        public string SessionTicket { get; private set; }
+        public static EntityKey EntityToken { get; private set; }
+        public PlayFabAuthenticationContext AuthenticationContext { get; private set; }
+
+        private const string _LoginRememberKey = "PlayFabLoginRemember";
+        private const string _PlayFabRememberMeIdKey = "PlayFabIdPassGuid";
+        private const string _PlayFabAuthTypeKey = "PlayFabAuthType";
+
+        public bool RememberMe
+        {
+            get { return PlayerPrefs.GetInt(_LoginRememberKey, 0) != 0; }
+            set { PlayerPrefs.SetInt(_LoginRememberKey, value ? 1 : 0); }
+        }
+
+        /// <summary>
+        /// Remember the type of authenticate for the user
+        /// </summary>
+        public AuthTypes AuthType
```

Comment:
@MarcoWilliamsPF Looping back around to this:
Every interface function accepts an AuthTypes as a parameter.
I am concerned that this variable, and the preference both:

1. Doesn't serve a purpose
2. Conflicts with the AuthType being manually set in almost every input method.

I would opt to completely remove this property, as it is more flexible, and removes the conflict.